### PR TITLE
add map for previously found custom lookup values

### PR DESCRIPTION
**Pull request by jcollins-axway**

### CODE REVIEW

File: pkg/customunit/handler.go [hunk 1]
```diff
@@ -391,10 +394,37 @@ func (c *CustomUnitHandler) ManagedApplicationLookup(appLookup *customunits.AppL
...
+		c.logger.Error("not able to lookup application by custom attribute without the attribute name set")
+		return nil
+	}
+
+	customKey := fmt.Sprintf("%s_%s", appCustomAttr, appValue)
+	if apiID, ok := c.apiCustomLookups[customKey]; ok {
+		return c.cache.GetAPIServiceWithAPIID(apiID)
+	}
+
+	for _, key := range c.cache.GetAPIServiceKeys() {
```
Comment (kasuvy) [inline pkg/customunit/handler.go:411]: Should this be GetManagedApplicationCacheKeys() ?

File: pkg/customunit/handler.go [hunk 2]
```diff
@@ -391,10 +394,37 @@ func (c *CustomUnitHandler) ManagedApplicationLookup(appLookup *customunits.AppL
...
+}
+
+func (c *CustomUnitHandler) customAppLookup(appValue, appCustomAttr string) *v1.ResourceInstance {
+	if appCustomAttr == "" {
+		c.logger.Error("not able to lookup application by custom attribute without the attribute name set")
+		return nil
+	}
+
+	customKey := fmt.Sprintf("%s_%s", appCustomAttr, appValue)
+	if apiID, ok := c.apiCustomLookups[customKey]; ok {
```
Comment (kasuvy) [inline pkg/customunit/handler.go:407]: should this be appCustomLookups[] ?

File: pkg/customunit/handler.go [hunk 3]
```diff
@@ -391,10 +394,37 @@ func (c *CustomUnitHandler) ManagedApplicationLookup(appLookup *customunits.AppL
...
+		return c.cache.GetAPIServiceWithAPIID(apiID)
```
Comment (kasuvy) [inline pkg/customunit/handler.go:408]: this returns APIService ?

File: pkg/customunit/handler.go (hunk 2 above)
Comment (jcollins-axway) [reply pkg/customunit/handler.go:407]: yessir

File: pkg/customunit/handler.go (hunk 1 above)
Comment (jcollins-axway) [reply pkg/customunit/handler.go:411]: yessir

File: pkg/customunit/handler.go (hunk 3 above)
Comment (jcollins-axway) [reply pkg/customunit/handler.go:408]: fixing

**Review by sbolosan — APPROVED** [review]

**Review by vivekschauhan — APPROVED** [review]

**Review by sbolosan — APPROVED** [review]

**Review by kasuvy — APPROVED** [review]

**Pull request merged**